internal/trace/transport.go

```diff
@@ -0,0 +1,74 @@
+// Copied and adapted from oras (https://github.com/oras-project/oras)
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+Copied and adapted from oras (https://github.com/oras-project/oras)
+*/
+
+package trace
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/notaryproject/notation-go/log"
+)
+
+// Transport is an http.RoundTripper that keeps track of the in-flight
+// request and add hooks to report HTTP tracing events.
+type Transport struct {
+	http.RoundTripper
+}
+
+func NewTransport(base http.RoundTripper) *Transport {
+	return &Transport{base}
+}
+
+// RoundTrip calls base roundtrip while keeping track of the current request.
+func (t *Transport) RoundTrip(req *http.Request) (resp *http.Response, err error) {
+	ctx := req.Context()
+	e := log.GetLogger(ctx)
+
+	e.Debugf(" Request URL: %q", req.URL)
+	e.Debugf(" Request method: %q", req.Method)
+	e.Debugf(" Request headers:")
+	logHeader(req.Header, e)
+
+	resp, err = t.RoundTripper.RoundTrip(req)
+	if err != nil {
+		e.Errorf("Error in getting response: %w", err)
+	} else if resp == nil {
+		e.Errorf("No response obtained for request %s %q", req.Method, req.URL)
+	} else {
+		e.Debugf(" Response Status: %q", resp.Status)
+		e.Debugf(" Response headers:")
```

Comment:
```suggestion
		e.Debugf("< Response status: %q", resp.Status)
		e.Debugf("< Response headers:")
```

/cc @qweeah to check if it should be applied to oras.

Reply:
Should directional hints also be added to each line of request output? e.g. below is an example of curl
```
$ curl mcr.microsoft.com -v
*   Trying 204.79.197.219:80...
* TCP_NODELAY set
* Connected to mcr.microsoft.com (204.79.197.219) port 80 (#0)
> GET / HTTP/1.1
> Host: mcr.microsoft.com
> User-Agent: curl/7.68.0
> Accept: */*
>
* Mark bundle as not supporting multiuse
< HTTP/1.1 308 Permanent Redirect
< Location: https://mcr.microsoft.com/
< X-MSEdge-Ref: Ref A: A52A80F8316549B5B0671E55D1A02D90 Ref B: SG2EDGE1621 Ref C: 2022-12-05T02:07:05Z
< Date: Mon, 05 Dec 2022 02:07:04 GMT
< Content-Length: 0
<
* Connection #0 to host mcr.microsoft.com left intact
```

Reply:
Updated.